Improve Getting Started vignette (#234)
epiverse-trace/epichains: vignettes/epichains.Rmd

)
```

This vignette demonstrates how get started with using the _epichains_ package
for simulating transmission chains or estimating the likelihood of observing
a transmission chain.

::: {.alert .alert-info}
* To understand the theoretical background of the branching processes methods
used here, refer to the theory vignette `vignette("theoretical_background")`.
* To understand the software development design decisions and implementation details of the package,
refer to the design vignette `vignette("design-principles")`.
:::

> Epidemics spread through populations when infected individuals transmit the

Reviewer: Edited to distinguish between point source outbreaks etc.
```suggestion
Infectious disease epidemics can spread through populations when a chain of infected individuals transmit the
```

* To understand the software development design decisions and implementation details of the package,
refer to the [design vignette](https://epiverse-trace.github.io/epichains/articles/design-principles.html)
:::

Epidemics spread through when infected individuals transmit the
infection to others. If we assume that each case infects a random number of
other individuals (the offspring) through time units called generations,
and that this transmission process is governed by a reproduction number and
a probability distribution describing how the offspring are produced through
the time unit, then the epidemic can be modelled as a branching process.

Branching processes are a stochastic process where each individual in a
generation gives rise to a random number of individuals in the next generation.
The distribution of the number of offspring is called the offspring
distribution.

The size of the transmission chain is the total number of
individuals infected by a single case, and the length of the transmission
chain is the number of generations from the first case to the last case they
produced before the chain ended (See figure below).

> ```{r out.width = "75%", echo = FALSE, fig.cap = "An example of a transmission chain starting with a single case C1. Cases are represented by blue circles and arrows indicate who infected whom. The chain grows through generations G1, G2, and G3, producing cases C2, C3, C4, C5, C6, and C7. The chain ends at generation G3 with case C7. The size of C1's chain is 7 (sum of all blue circles) and the length is 3 (maximum number of generations reached by C1's chain).", fig.alt = "The figure shows an example of a transmission chain starting with a single case C1. The figure is depicted as blue circles linked by orange directed arrows showing the cases they produced. Each generation is marked by a brown rectangle and labelled Gen 1, Gen 2, and Gen 2. The chain grows through generations G1, G2, and G3. Case C1 is Gen 1 and produces cases C2, and C3 in generation 2. In Gen 3, C2 produces C4 and C5, and C3 produces C7. The chain ends in Gen 3. The chain size of C1 is 7 (that is the sum of all the blue circles, representing cases) and the length is 3 (maximum number of generations reached by C1's chain)."}

Author: > The generations are labelled "Gen X" in the figure and "GX" here - suggest to streamline. Also there's no C7 - is the image old version of the figure perhaps?

Thanks for catching that. Yes, I updated the figure later on and forgot to update the text. See change in 6161d8350f15eb4b75c0171a71b01914ded2d428.

> For the outbreak size / length, bear in mind that the index case is not counted (so the figure is really size 5 / length 2).

I think we do count the index case. See https://github.com/epiverse-trace/bpmodels/blob/main/R/simulate.r#L190-L195. `stat_track` starts off as 1 for each seeded case and is incremented based on `size` or `length`. Using bpmodels code because this part was converted into a helper function here so will require navigating between two places.

> ```{r out.width = "75%", echo = FALSE, fig.cap = "An example of a transmission chain starting with a single case C1. Cases are represented by blue circles and arrows indicate who infected whom. The chain grows through generations G1, G2, and G3, producing cases C2, C3, C4, C5, C6, and C7. The chain ends at generation G3 with case C7. The size of C1's chain is 7 (sum of all blue circles) and the length is 3 (maximum number of generations reached by C1's chain).", fig.alt = "The figure shows an example of a transmission chain starting with a single case C1. The figure is depicted as blue circles linked by orange directed arrows showing the cases they produced. Each generation is marked by a brown rectangle and labelled Gen 1, Gen 2, and Gen 2. The chain grows through generations G1, G2, and G3. Case C1 is Gen 1 and produces cases C2, and C3 in generation 2. In Gen 3, C2 produces C4 and C5, and C3 produces C7. The chain ends in Gen 3. The chain size of C1 is 7 (that is the sum of all the blue circles, representing cases) and the length is 3 (maximum number of generations reached by C1's chain)."}

Reviewer 2: Ah yes you're right, sorry.

> ```{r out.width = "75%", echo = FALSE, fig.cap = "An example of a transmission chain starting with a single case C1. Cases are represented by blue circles and arrows indicate who infected whom. The chain grows through generations G1, G2, and G3, producing cases C2, C3, C4, C5, C6, and C7. The chain ends at generation G3 with case C7. The size of C1's chain is 7 (sum of all blue circles) and the length is 3 (maximum number of generations reached by C1's chain).", fig.alt = "The figure shows an example of a transmission chain starting with a single case C1. The figure is depicted as blue circles linked by orange directed arrows showing the cases they produced. Each generation is marked by a brown rectangle and labelled Gen 1, Gen 2, and Gen 2. The chain grows through generations G1, G2, and G3. Case C1 is Gen 1 and produces cases C2, and C3 in generation 2. In Gen 3, C2 produces C4 and C5, and C3 produces C7. The chain ends in Gen 3. The chain size of C1 is 7 (that is the sum of all the blue circles, representing cases) and the length is 3 (maximum number of generations reached by C1's chain)."}

Author: Should we not count the seeded cases? We could just subtract 1 from the resulting vector.

> ```{r out.width = "75%", echo = FALSE, fig.cap = "An example of a transmission chain starting with a single case C1. Cases are represented by blue circles and arrows indicate who infected whom. The chain grows through generations G1, G2, and G3, producing cases C2, C3, C4, C5, C6, and C7. The chain ends at generation G3 with case C7. The size of C1's chain is 7 (sum of all blue circles) and the length is 3 (maximum number of generations reached by C1's chain).", fig.alt = "The figure shows an example of a transmission chain starting with a single case C1. The figure is depicted as blue circles linked by orange directed arrows showing the cases they produced. Each generation is marked by a brown rectangle and labelled Gen 1, Gen 2, and Gen 2. The chain grows through generations G1, G2, and G3. Case C1 is Gen 1 and produces cases C2, and C3 in generation 2. In Gen 3, C2 produces C4 and C5, and C3 produces C7. The chain ends in Gen 3. The chain size of C1 is 7 (that is the sum of all the blue circles, representing cases) and the length is 3 (maximum number of generations reached by C1's chain)."}

Reviewer 2: I think you could argue either way - I find it clearer to include them i.e. as it is at the moment. But perhaps there's an opportunity here to clarify this here, i.e. add "(including the index case)" somewhere in the sentence before the fig. caption (l.51).

> ```{r out.width = "75%", echo = FALSE, fig.cap = "An example of a transmission chain starting with a single case C1. Cases are represented by blue circles and arrows indicate who infected whom. The chain grows through generations G1, G2, and G3, producing cases C2, C3, C4, C5, C6, and C7. The chain ends at generation G3 with case C7. The size of C1's chain is 7 (sum of all blue circles) and the length is 3 (maximum number of generations reached by C1's chain).", fig.alt = "The figure shows an example of a transmission chain starting with a single case C1. The figure is depicted as blue circles linked by orange directed arrows showing the cases they produced. Each generation is marked by a brown rectangle and labelled Gen 1, Gen 2, and Gen 2. The chain grows through generations G1, G2, and G3. Case C1 is Gen 1 and produces cases C2, and C3 in generation 2. In Gen 3, C2 produces C4 and C5, and C3 produces C7. The chain ends in Gen 3. The chain size of C1 is 7 (that is the sum of all the blue circles, representing cases) and the length is 3 (maximum number of generations reached by C1's chain)."}

Author: I've clarified that here 8127f57b0e82f4fe4ae2990968f22a1783295be3.

1. Data on observed transmission chains summaries (sizes or lengths).

#### What we assume {-}

1. An offspring distribution that governs the number of secondary cases
produced by each case.
2. Parameters of the offspring distribution (for example, mean and overdispersion if using a negative binomial; which can then be interpreted as reproduction number and superspreading coefficient, respectively)
3. (optional) An observation process/probability that generates the observed chain
summaries.
4. (optional) A threshold of chain summaries that are considered too large to be consistent with a reproduction number of $R<1$ (e.g., $N$ cases or $T$ generations in total).
:::

### [`likelihood()`](https://epiverse-trace.github.io/epichains/reference/likelihood.html)

This function calculates the likelihood/loglikelihood of observing a vector of
outbreak summaries obtained from transmission chains. "Outbreak summaries" here
refer to transmission chain sizes or lengths/durations.

`likelihood()` requires a vector of chain summaries (sizes or lengths),
`chains`, the corresponding statistic to use, `statistic`, the offspring
distribution, `offspring_dist` and its associated parameters.

> `offspring_dist` is specified as the function that is used to generate random

Reviewer: ```suggestion
`offspring_dist` is specified by the R function that is used to generate random
```

numbers, i.e. `rpois` for Poisson, `rnbinom` for negative binomial, or a custom
function.

By default, the result is a log-likelihood but if `log = FALSE`, then
likelihoods are returned (See `?likelihood` for more details).

::: {.alert .alert-info}
To understand how `likelihood()` works see the section on [How `likelihood()` works](#how-likelihood-works).
:::

Let's look at the following example where we estimate the log-likelihood of
observing `chain_sizes`.
```{r estimate_likelihoods}
set.seed(121)
# example of observed chain sizes
# randomly generate 20 chains of size between 1 to 10
chain_sizes <- sample(1:10, 20, replace = TRUE)
chain_sizes

> # estimate loglikelihood of the observed chain sizes

Reviewer: ```suggestion
# estimate loglikelihood of the observed chain sizes for given lambda
```

library("epichains")
```

## Transmission chains likelihoods

::: {.alert .alert-primary}
### Use case {-}

> Suppose we have observed a number of outbreak clusters, each arising from a

Author: Thanks. I've changed them to "chains" for consistency here fe2c653cb3ab7c4b2ed0a153ada045d1f2c28ce1.

chain_sizes <- sample(1:10, 20, replace = TRUE)
# get their likelihood
liks <- likelihood(
  chains = chain_sizes,
  offspring_dist = rbinom,
  statistic = "size",
  size = 1,
  prob = 0.9,
  nsim_offspring = 250
)
liks
```

## Transmission chains simulation

::: {.alert .alert-primary}
### Use case {-}
We want to simulate a scenario where a number of outbreak clusters are each
produced by a separate external case. We want to simulate the chains of
transmission that result from these cases, given a specific offspring
distribution and a reproduction number.
:::

::: {.alert .alert-secondary}
#### What we have {-}

1. An initial number of cases that seed the outbreak.

#### What we assume {-}

1. An offspring distribution that governs the number of secondary cases
produced by each case.
2. A reproduction number that governs the average number of secondary cases
produced by each case.
3. An observation process/probability that generates the observed chain
summaries.
4. A threshold of chain summaries that are considered too large for the given
transmission properties. For example, we do not expect each case to produce

> more than $N$ cases or last more than $T$ generations.

Author: I've fixed that here d9e0cab9fb5ae3a41262ad2832b9dafdcfbc3cef.
